controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java

```diff
@@ -3623,4 +3623,7 @@ OperationFailedRuntimeException capabilityAlreadyRegisteredInContext(String capa
 
     @Message(id = 475, value = "Value for attribute '%s' is invalid.")
     OperationFailedException invalidAttributeValue(String attributeName);
+
+    @Message(id = 476, value = "Unsupported usage of expression:")
```

Comment:
I think this needs a bit more info to explain things, as now it's going to output a capability name, which the user may not understand. The name will include an expression that perhaps they will recognize.

Perhaps

"Required capabilities are not available. This likely is due to the use of an expression string in a configuration attribute that does not support expressions:"

Hmm, but thinking further... with what is in the current code we have, for example:

```
Required capabilities are not available:
     some.capability.name; Possible registration points for this capability:
          blahblahblah
     some.other.capability.name;  There are no known registration points which can provide this capability.
```

So shouldn't the result here be:

```
Required capabilities are not available:
     some.capability.name; Possible registration points for this capability:
          blahblahblah
     some.other.capability.name; There are no known registration points which can provide this capability.
     another.capability.${name}; This unresolvable capability likely is due to the use of an expression string in a configuration attribute that does not support expressions.
```

The appendPossibleProviderPoints method could handle that. If there are no possible provider points, report that, as it means something is odd beyond the use of the expression. Otherwise if there's an expression use the "does not support expressions" message. Otherwise report the provider points.